### PR TITLE
rewrite html() methods and template tags for efficiency

### PR DESCRIPTION
**Pull request by nschneid**

### CODE REVIEW

File: wiki/plugins/metadata/models.py [hunk 1]
```diff
@@ -791,13 +821,15 @@ class CorpusSentence(models.Model):
...
 
-    def get_absolute_url(self):
+    @cached_property
+    def url(self):
         return reverse('wiki:corpus_sentence_view', args=[self.language.slug, self.corpus, self.sent_id])
 
+    @cached_property
     def html(self):
-        return format_html('<a href="{}" class="corpussentence">{}</a>', self.get_absolute_url(), self.sent_id)
+        return format_html(f'<a href="{self.url}" class="corpussentence">{{}}</a>', self.sent_id)
```
Comment (ablodge) [inline wiki/plugins/metadata/models.py]: Could this line be `format_html(f'<a href="{self.url}" class="corpussentence">{self.sent_id}</a>') ` ?

File: wiki/plugins/metadata/models.py [hunk 2]
```diff
@@ -858,21 +890,23 @@ def exnum(self):
...
-    def get_absolute_url(self):
+    @cached_property
+    def url(self):
         return reverse('wiki:ptoken_view', args=[self.exnum])
 
+    @cached_property
     def html(self):
         """Linked example number in parentheses"""
-        return format_html('<a href="{}" class="exnum">({})</a>', self.get_absolute_url(), self.exnum)
+        return format_html(f'<a href="{self.url}" class="exnum">({{}})</a>', self.exnum)
```
Comment (ablodge) [inline wiki/plugins/metadata/models.py]: Same as above.

File: wiki/plugins/metadata/tables.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ class PTokenAnnotationTable(tables.Table):
     sentid = tables.Column(accessor='sentence', verbose_name='Sent ID')
     
     def render_exid(self, record):
-        return record.html()
+        return record.html
```
Comment (ablodge) [inline wiki/plugins/metadata/tables.py]: Starting here, you switched to returning a function? Can you explain this line?

**Comment by ablodge** [comment]
Error:
```
NoReverseMatch at /
'djdt' is not a registered namespace

```
Is there a package I need to install?

**Comment by ablodge** [comment]
The above error is because I'm working on dev mode and you're working on base mode. If I switch to base, everything works.

File: wiki/plugins/metadata/models.py (hunk 1 above)
Comment (nschneid) [reply wiki/plugins/metadata/models.py]: No, the difference is that `format_html` escapes the arguments before inserting them into the pattern. I believe the URL should already be properly escaped, but the sentence ID could have special characters.

**Comment by nschneid** [comment]
No, I'm in dev mode and I don't get that error. 'djdt' is from the debug toolbar; I haven't merged your recent changes for profiling.

File: wiki/plugins/metadata/tables.py (hunk 3 above)
Comment (nschneid) [reply wiki/plugins/metadata/tables.py]: In models.py, `CorpusSentence.html` has been made a property (via the `@cached_property` decorator).

**Pull request merged**